presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTimestamp.java

```diff
@@ -67,6 +84,12 @@ public boolean equals(Object obj)
     @Override
     public String toString()
     {
-        return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(sessionTimeZoneKey.getId())).format(formatter);
+        if (isDeprecatedTimestamp()) {
+            return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(sessionTimeZoneKey.get().getId())).format(formatter);
+        }
+        else {
+            // FIXME this should be exact equivalent of TIMESTAMP to VARCHAR cast
```

Comment:
This FIXME isn't resolved in this PR. intentional?

Reply:
Also, this is `@JsonValue` method, so if there is anything to be fixed here, it needs to be fixed.

Reply:
No idea what is to be done here. It looks ok. Removing comment.